pkg/controller/machineset/controller.go

```diff
@@ -183,9 +184,9 @@ func (r *ReconcileMachineSet) Reconcile(request reconcile.Request) (reconcile.Re
 	updatedMS, err := updateMachineSetStatus(r.Client, machineSet, newStatus)
 	if err != nil {
 		if syncErr != nil {
-			return reconcile.Result{}, fmt.Errorf("failed to sync machines. %v. failed to update machine set status. %v", syncErr, err)
+			return reconcile.Result{}, errors.Wrapf(err, "failed to sync machines: %v. failed to update machine set status", syncErr)
```

Comment:
so this is a case that the Wrap functions do not solve - returning a combination of two errors.
but the real problem is that this function should be returning a list of errors, ideally, because it does have such an edge case.

i think this change is mostly OK (ish).